packages/eslint-plugin/typings/eslint-rules.d.ts

```diff
@@ -641,6 +641,28 @@ declare module 'eslint/lib/rules/init-declarations' {
   export = rule;
 }
 
+declare module 'eslint/lib/rules/no-invalid-this' {
+  import { TSESLint, TSESTree } from '@typescript-eslint/experimental-utils';
+
+  const rule: TSESLint.RuleModule<
+    'unexpectedThis',
```

Comment:
you probably had a look at the `master` source code for the rule.
eslint master is currently prepped for their unreleased v7 version.

You need to instead look at the current version's code using GH tag's
https://github.com/eslint/eslint/blob/v6.8.0/lib/rules/no-invalid-this.js

The rule had no message IDs in 6.8.0

Reply:
oh, shoot.

I completly forgot about the version about this. thanks a lot for pointing. 

I will change it

Reply:
i think `messageId` is required for tests. 

![image](https://user-images.githubusercontent.com/26347874/80729606-1ecd2380-8b26-11ea-934f-74d85ded6e21.png)


here is what I changed for the rule module

```ts
declare module 'eslint/lib/rules/no-invalid-this' {
  import { TSESLint, TSESTree } from '@typescript-eslint/experimental-utils';

  const rule: TSESLint.RuleModule<
    never,
    [
      {
        capIsConstructor?: boolean;
      }?,
    ],
    {
      Program(node: TSESTree.Program): void;
      'Program:exit'(node: TSESTree.Program): void;
      FunctionDeclaration(node: TSESTree.FunctionDeclaration): void;
      'FunctionDeclaration:exit'(node: TSESTree.FunctionDeclaration): void;
      FunctionExpression(node: TSESTree.FunctionExpression): void;
      'FunctionExpression:exit'(node: TSESTree.FunctionExpression): void;
      ThisExpression(node: TSESTree.ThisExpression): void;
    }
  >;
  export = rule;
}
```

and having `message` in errors object will give type error 
![image](https://user-images.githubusercontent.com/26347874/80729910-84211480-8b26-11ea-9187-57e6cc5b9685.png)


as  `message is not a property in `TestCaseError`. 

should I add `message` in `TestCaseError` and add `options ?` for `messageId` ?

Reply:
nah - message is purposely excluded from the types to force people to use message ids.

You can workaround this with an `any` like other extension rules have:

https://github.com/typescript-eslint/typescript-eslint/blob/188b68979d114ba0957a076fe9c27d154223512c/packages/eslint-plugin/tests/rules/no-useless-constructor.test.ts#L13-L18

https://github.com/typescript-eslint/typescript-eslint/blob/026ceb991d12a7bf0f37cefee009bf2f408153ec/packages/eslint-plugin/tests/eslint-rules/no-restricted-globals.test.ts#L54-L99

https://github.com/typescript-eslint/typescript-eslint/blob/188b68979d114ba0957a076fe9c27d154223512c/packages/eslint-plugin/tests/rules/quotes.test.ts#L18-L34

Myself and some others have PR'd messageIds into old rules in eslint core, so with the 7.0 release we'll be able to remove this unsafety.

Reply:
ok I will try this 👍